src/main/apisix/templates/configmap.yaml

```diff
@@ -0,0 +1,159 @@
+      #     - 9101
+      #   udp:                        # UDP proxy port list
+      #     - 9200
+      #     - 9211
+      # dns_resolver:
+      #   {{- range $resolver := .Values.dns.resolvers }}
+      #   - {{ $resolver }}
+      #   {{- end }}
+      dns_resolver_valid: {{.Values.dns.validity}}
+      resolver_timeout: {{.Values.dns.timeout}}
+      ssl:
+        enable: {{ .Values.gateway.tls.enabled }}
+        enable_http2: {{ .Values.gateway.tls.http2.enabled }}
+        listen_port: {{ .Values.gateway.tls.containerPort }}
+        ssl_protocols: "TLSv1 TLSv1.1 TLSv1.2 TLSv1.3"
+        ssl_ciphers: "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA256:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA"
+
+    nginx_config:                     # config for render the template to genarate nginx.conf
+      error_log: "/dev/stderr"
+      error_log_level: "warn"         # warn,error
+      worker_processes: auto
+      worker_rlimit_nofile: 20480     # the number of files a worker process can open, should be larger than worker_connections
+      worker_shutdown_timeout: 240s   # timeout for a graceful shutdown of worker 
+      event:
+        worker_connections: 10620
+      http:
+        access_log: "/dev/stdout"
+        keepalive_timeout: 60s         # timeout during which a keep-alive client connection will stay open on the server side.
+        client_header_timeout: 60s     # timeout for reading client request header, then 408 (Request Time-out) error is returned to the client
+        client_body_timeout: 60s       # timeout for reading client request body, then 408 (Request Time-out) error is returned to the client
+        send_timeout: 10s              # timeout for transmitting a response to the client.then the connection is closed
+        underscores_in_headers: "on"   # default enables the use of underscores in client request header fields
+        real_ip_header: "X-Real-IP"    # http://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_header
+        real_ip_from:                  # http://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from
+          - 127.0.0.1
+          - 'unix:'
+        #lua_shared_dicts:              # add custom shared cache to nginx.conf
+        #  ipc_shared_dict: 100m        # custom shared cache, format: `cache-key: cache-size`
+
+    etcd:
+      host:
+      {{- if .Values.etcd.enabled }}
+        - "http://{{ include "apisix.fullname" . }}-etcd-0.{{ include "apisix.fullname" . }}-etcd:2379"
```

Comment:
# Observations

I deployed and got error "etcd service could not be resolved":

```
root@i-md5d3cv8:~# kubectl -n apps logs -f api-apisix-f4dd54bd8-plqkl
2020/08/26 07:38:14 [error] 16#16: *36 [lua] heartbeat.lua:89: failed to fetch etcd version: api-apisix-etcd-0.api-apisix-etcd could not be resolved (3: Host not found), context: ngx.timer
2020/08/26 07:38:19 [error] 16#16: *36 [lua] heartbeat.lua:117: failed to report heartbeat information: timeout, context: ngx.timer
```

![image](https://user-images.githubusercontent.com/4861515/91285448-d0701e80-e7bf-11ea-86ad-fbf00aad4dc5.png)

I tried to set `etcd.host` to `http://{{ .Release.Name }}-etcd:2379` and it still didn't work. Seems Lua has issues with DNS resolving. If I execute `ping api-etcd` within the pod, it can resolve the address correctly:

```
[root@api-apisix-9c465d549-2db4f apisix]# ping api-etcd
PING api-etcd.apps.svc.cluster.local (10.10.96.137) 56(84) bytes of data.
64 bytes from api-etcd-0.api-etcd.apps.svc.cluster.local (10.10.96.137): icmp_seq=1 ttl=62 time=0.644 ms
64 bytes from api-etcd-0.api-etcd.apps.svc.cluster.local (10.10.96.137): icmp_seq=2 ttl=62 time=0.451 ms
```

# Suggestions

1. Investigate Lua solutions (prefer this one)
2. Use FQDN for etcd service, i.e. {{ .Release.Name}}-etcd.{{ namespace }}.svc.{{ cluster domain}}

Reply:
let me check

Reply:
I think we need update to the latest version of APISIX .

Reply:
fixed